spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js

```diff
@@ -209,6 +209,92 @@ describe('GA4 link tracker', function () {
     })
   })
 
+  describe('different types of clicks', function () {
+    beforeEach(function () {
+      attributes = {
+        event_name: 'navigation',
+        type: 'a link',
+        index_link: 1
+      }
+      expected.event = 'event_data'
+      expected.event_data.type = 'a link'
+      expected.event_data.external = 'false'
+      expected.event_data.index = {
+        index_link: 1,
+        index_section: undefined,
+        index_section_count: undefined
+      }
+      var link = '#aNormalLink'
+      element = document.createElement('a')
+      element.setAttribute('data-ga4-link', JSON.stringify(attributes))
+      element.setAttribute('href', link)
+      var linkText = 'A normal link'
+      element.textContent = linkText
+      expected.event_data.text = linkText
+      expected.event_data.url = link
+      expected.event_data.link_path_parts['1'] = link
+    })
+
+    it('tracks primary clicks on a single link', function () {
+      expected.event_data.method = 'primary click'
+      initModule(element, true)
+      expect(window.dataLayer[0]).toEqual(expected)
```

Comment:
Shouldn't this test dispatch a click event? I'm confused - it's clearly passing, though.

Reply:
@andysellick I used the `initModule` function here, which when passed `true` clicks the element that is passed through

Reply:
Ahhh of course. Nice.